lib/cookie.c

```diff
@@ -217,6 +240,49 @@ static bool pathmatch(const char *cookie_path, const char *request_uri)
   return ret;
 }
 
+/*
+ * Return the top-level domain, for optimal hashing.
+ */
+static const char *get_top_domain(const char * const domain, size_t *outlen)
+{
+  size_t len;
+  const char *first, *last;
+
+  if(!domain)
+    return NULL;
+
+  len = strlen(domain);
+  first = memchr(domain, '.', len);
+  last = memrchr(domain, '.', len);
+
+  if(outlen)
+    *outlen = len;
+
+  if(first == last)
+    return domain;
+
+  first = memrchr(domain, '.', (size_t)(last - domain - 1));
```

Comment:
Why -1 ? This will miss character before the last dot, causing `blabla..org` to set first = NULL